src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/ExponentialMovingAverageCounter.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Exponential moving average counter.
+    /// </summary>
+    internal class ExponentialMovingAverageCounter
+    {
+        /// <summary>
+        /// Average value of the counter.
+        /// </summary>
+        private double? average;
+
+        /// <summary>
+        /// Value of the counter during current interval of time.
+        /// </summary>
+        private long current;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExponentialMovingAverageCounter"/> class.
+        /// </summary>
+        /// <param name="coefficient">Exponential coefficient.</param>
+        public ExponentialMovingAverageCounter(double coefficient)
+        {
+            this.Coefficient = coefficient;
+        }
+
+        /// <summary>
+        /// Gets exponential coefficient (must be between 0 and 1).
+        /// </summary>
+        public double Coefficient { get; private set; }
```

Comment:
And if the check fails we'd do what? Throwing isn't a good idea (failure just in the monitoring code potentially preventing app from executing entirely). If it becomes (reported) to be a problem we can add some checking (say, in debug code only or something).